com.unity.shadergraph/Editor/Drawing/MaterialEditor/ShaderGraphPropertyDrawers.cs

```diff
@@ -0,0 +1,299 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEditor;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Object = UnityEngine.Object;
+
+namespace UnityEditor.ShaderGraph.Drawing
+{
+    internal static class ShaderGraphPropertyDrawers
+    {
+        static Dictionary<MinimalCategoryData.GraphInputData, bool> s_CompoundPropertyFoldoutStates = new Dictionary<MinimalCategoryData.GraphInputData, bool>();
+
+        public static void DrawShaderGraphGUI(MaterialEditor materialEditor, MaterialProperty[] properties)
+        {
+            Material m = materialEditor.target as Material;
+            Shader s = m.shader;
+            string path = AssetDatabase.GetAssetPath(s);
+            ShaderGraphMetadata metadata = null;
+            foreach (var obj in AssetDatabase.LoadAllAssetsAtPath(path))
```

Comment:
Hey Felipe, good question, so I threw in a breakpoint to test this, and it looks like this gets called whenever the Material Inspector triggers its own internal update cycle, which appears to be when:
- The user clicks on a Material asset derived from a shader graph (triggering the material inspector)
- The user moves their mouse across an input field 
- Whenever a shader graph is saved, changing the Material asset on disk (requiring a refresh of the inspector)

Reply:
I'm afraid this might cause editor to hang/hiccups for large projects. Imo you should not do this in the situations above. 
I won't block this PR due to that but this needs more testing/fixing.

Reply:
Hey Felipe, this is a good point, though the actual data that we are loading is fairly light, and is not the entire graph data itself as opposed to a slimmed down version that contains the minimal data needed to draw the categories.

I'm asking @LandonTownsendUnity to test out the categories in a large URP project to see if this has any visible impact to the UX or otherwise slows down the project.